File: typescript/base-hmi-types/src/screens/widgets/HmiButton.ts
import { HmiProperty } from "../base.js";
import { HmiButtonBase } from "./HmiButtonBase.js";
import { HmiButtonType } from "./HmiButtonType.js";
import { HmiObjectType } from "../base/HmiObjectType.js";

export class HmiButton extends HmiButtonBase {
  constructor() {
    super();
    this.hmiObjectType = HmiObjectType.HmiButton;
  }

  mode?: HmiProperty<HmiButtonType>;
}

File: src/BaseHmiTypes/Screens/Widgets/HmiButton.cs
using BaseHmiTypes.Screens.Base;

namespace BaseHmiTypes.Screens.Widgets;

public class HmiButton : HmiButtonBase
{
    public HmiButton()
    {
        HmiObjectType = BaseHmiTypes.Screens.Base.HmiObjectType.HmiButton;
    }

    public HmiProperty<HmiButtonType>? Mode { get; set; }
}
